### PR TITLE
[Android] Enable proguard for crosswalk to compress apk size

### PR DESCRIPTION
**Pull request by heke123**
Add flag --enable-proguard, disable by default. Only works with
Embedded mode.  Approximately shrinked 150k apk size(tested in x86)

BUG=XWALK-3854

### CODE REVIEW

**Comment by halton** [comment]
@heke123 could you add SIZE reduced info in git log?

File: app/android/app_template/proguard-xwalk.txt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+-keepattributes InnerClasses
+-keep class **.R
+-keep class **.R$* {
+    <fields>;
+}
+
+-adaptresourcefilenames    **.properties,**.gif,**.jpg
+-adaptresourcefilecontents **.properties,META-INF/MANIFEST.MF
+
+### keep native & callbacks
```
Comment (halton) [inline app/android/app_template/proguard-xwalk.txt]: nit, comment should capital for the first letter.

File: app/android/app_template/proguard-xwalk.txt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+-keep class org.chromium.base.*Native*
+-keep class org.chromium.base.JNINamespace
+-keepclasseswithmembers class org.chromium.** {
+    @org.chromium.base.AccessedByNative <fields>;
+}
+-keepclasseswithmembers class org.chromium.** {
+    @org.chromium.base.*Native* <methods>;
+}
+
+# TODO(mef) remove unnecessary classes from base, so we don't have to preserve
```
Comment (halton) [inline app/android/app_template/proguard-xwalk.txt]: mef, who is that guy?

File: app/tools/android/make_apk.py [hunk 3]
```diff
@@ -341,6 +341,23 @@ def Execution(options, app_info):
...
   RunCommand(update_project_cmd)
 
+  # Enable proguard
+  if options.mode == 'embedded' and options.enable_proguard:
+    print(' * Enabling proguard config files')
+    # Enable proguard in project.properies.
+    if not os.path.exists(os.path.join(app_dir,'project.properties')):
+      print('Error, project.properties file not found!')
+      sys.exit(14)
+    file_prop = file(os.path.join(app_dir,'project.properties'),'a')
```
Comment (halton) [inline app/tools/android/make_apk.py]: space before comma

File: app/tools/android/make_apk.py [hunk 4]
```diff
@@ -416,6 +433,17 @@ def Execution(options, app_info):
   shutil.copyfile(src_file, dst_file)
   print(' (Location: %s)' % dst_file)
 
+  #Copy proguard dumping files
+  if options.mode == 'embedded' and options.enable_proguard\
```
Comment (halton) [inline app/tools/android/make_apk.py]: I do not think the backslash is needed.

File: app/tools/android/make_apk.py [hunk 5]
```diff
@@ -416,6 +433,17 @@ def Execution(options, app_info):
...
+    and not options.project_dir:
```
Comment (halton) [inline app/tools/android/make_apk.py]: 4-space for unfinished line

File: app/tools/android/make_apk.py [hunk 6]
```diff
@@ -416,6 +433,17 @@ def Execution(options, app_info):
...
+    proguard_dir = os.path.join(app_dir, 'bin/proguard/');
+    if os.path.exists(proguard_dir):
+      for afile in os.listdir(proguard_dir):
+        if afile.endswith('.txt'):
+          shutil.copy(os.path.join(proguard_dir,afile),xwalk_dir)
```
Comment (halton) [inline app/tools/android/make_apk.py]: ditto

**Comment by halton** [comment]
LGTM

**Comment by halton** [comment]
@heke123 please rebase your code so that I can merge.

**Comment by heke123** [comment]
@halton,  code rebase finished. thanks.

**Comment by halton** [comment]
Thanks, I'll wait until the intent to implement get approved.

**Comment by halton** [comment]
Merge since there are two lgtm for the intent to implement. Thanks @kenchris

**Pull request merged**